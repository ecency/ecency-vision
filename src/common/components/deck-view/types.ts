export interface DeckModel {
  data: any;
  dataParams: any[];
  header: {
    title: string;
    icon: JSX.Element | null;
  };
  listItemComponent: any;
  createdAt: Date;
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}
