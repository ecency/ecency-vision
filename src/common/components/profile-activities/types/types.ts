export interface ActivityTypes {
    type: string;
      parent_author?: string;
      parent_permlink?: string;
      author?: string;
      permlink?: string;
      payment?: number;
      payer?: string;
      following?: string;
      community?: string;
      approve?: boolean;
      proposal_ids?: string;
      account?: string;
      id?: string;
      what?: string[];
      voter?: string;
      witness?: string;
      json: string;
    timestamp: string;
  }