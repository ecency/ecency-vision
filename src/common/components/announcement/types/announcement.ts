export interface Announcement {
  id: number;
  title: string;
  description: string;
  button_text: string;
  button_link: string;
}

export interface LaterAnnouncement {
  id: number;
  dateTime: Date;
}
