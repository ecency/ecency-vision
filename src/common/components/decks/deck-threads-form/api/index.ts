export * from "./threads-api";
export * from "./use-community-api";
