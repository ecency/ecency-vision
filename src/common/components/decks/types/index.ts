export * from "./deck-grid-item";
export * from "./deck-grid";
export * from "./old-decks";
