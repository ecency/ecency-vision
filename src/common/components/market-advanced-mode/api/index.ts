export * from "./trading-view-api";
