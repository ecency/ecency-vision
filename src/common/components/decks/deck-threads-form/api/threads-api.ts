import { Entry } from "../../../../store/entries/types";
import { createReplyPermlink, makeJsonMetaDataReply } from "../../../../helper/posting";
import { comment } from "../../../../api/operations";
import tempEntry from "../../../../helper/temp-entry";
import { FullAccount } from "../../../../store/accounts/types";
import { version } from "../../../../../../package.json";
import { useMappedStore } from "../../../../store/use-mapped-store";

export function useThreadsApi() {
  const { activeUser, addReply, updateEntry } = useMappedStore();

  const request = async (entry: Entry, raw: string) => {
    if (!activeUser || !activeUser.data.__loaded) {
      throw new Error("No user");
    }

    const { author: parentAuthor, permlink: parentPermlink } = entry;
    const author = activeUser.username;
    const permlink = createReplyPermlink(entry.author);
    const tags = entry.json_metadata.tags || ["ecency"];

    const jsonMeta = makeJsonMetaDataReply(tags, version);

    await comment(author, parentAuthor, parentPermlink, permlink, "", raw, jsonMeta, null, true);

    const nReply = tempEntry({
      author: activeUser.data as FullAccount,
      permlink,
      parentAuthor,
      parentPermlink,
      title: "",
      body: raw,
      tags,
      description: null
    });

    // add new reply to store
    addReply(nReply);

    if (entry.children === 0) {
      // Activate discussion section with first comment.
      const nEntry: Entry = {
        ...entry,
        children: 1
      };

      updateEntry(nEntry);
    }

    return nReply;
  };

  return { request };
}
