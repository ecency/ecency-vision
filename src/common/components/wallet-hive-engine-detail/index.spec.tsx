import React from "react";

import Wallet from "./index";

import TestRenderer from "react-test-renderer";

import { createBrowserHistory } from "history";

import { initialState as transactionsInitialState } from "../../store/transactions/index";

import {
  globalInstance,
  dynamicPropsIntance1,
  fullAccountInstance,
  allOver
} from "../../helper/test-helper";
import { StaticRouter } from "react-router-dom";
import { FullAccount } from "../../store/accounts/types";

import HiveEngineToken from "../../helper/hive-engine-wallet";
import { DelegationEntry } from "../../api/hive-engine";
jest.mock("moment", () => () => ({
  fromNow: () => "in 5 days"
}));

let MOCK_MODE = 1;

jest.mock("../../api/hive", () => ({
  getConversionRequests: () =>
    new Promise((resolve) => {
      if (MOCK_MODE === 1) {
        resolve([]);
      }

      if (MOCK_MODE === 2) {
        resolve([
          {
            amount: "1200.000 HBD",
            conversion_date: "2021-01-29T21:19:54",
            id: 163785,
            owner: "fooo",
            requestid: 1040029198
          }
        ]);
      }
    })
}));

const account: FullAccount = {
  ...fullAccountInstance,
  name: "user1",
  balance: "0.000 HIVE",
  delegated_vesting_shares: "0.000000 VESTS",
  next_vesting_withdrawal: "2020-06-10T18:39:30",
  received_vesting_shares: "9916.938399 VESTS",
  reward_hbd_balance: "0.000 HBD",
  reward_vesting_hive: "10.207 HIVE",
  savings_balance: "0.001 HIVE",
  savings_hbd_balance: "0.000 HBD",
  hbd_balance: "447.514 HBD",
  to_withdraw: "91765607000000",
  vesting_shares: "83549915.208336 VESTS",
  vesting_withdraw_rate: "7058892.846154 VESTS",
  withdrawn: "56471142769232"
};

const defProps = {
  history: createBrowserHistory(),
  global: globalInstance,
  dynamicProps: dynamicPropsIntance1,
  users: [],
  activeUser: null,
  transactions: transactionsInitialState,
  account,
  signingKey: "",
  delegationList: [],
  tokenName: "POB",
  hiveEngineToken: new HiveEngineToken({
    symbol: "POB",
    name: "Proof of Brain",
    precision: 8,
    stake: "1000",
    balance: "100",
    delegationsIn: "400",
    delegationsOut: "0",
    icon: "",
    stakingEnabled: true,
    delegationEnabled: false
  }),
  addAccount: () => {},
  updateActiveUser: () => {},
  setSigningKey: () => {},
  fetchTransactions: () => {},
  fetchPoints: () => {},
  updateWalletValues: () => {},
  clearToken: () => {},
  openTransferDialog: () => {},
  closeTransferDialog: () => {},
  modifyTokenValues: () => {}
};

it("(1) Default Render", () => {
  const renderer = TestRenderer.create(
    <StaticRouter location="/" context={{}}>
      <Wallet {...defProps} />
    </StaticRouter>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(2) Render with converting HBD", async () => {
  MOCK_MODE = 2;

  const renderer = TestRenderer.create(
    <StaticRouter location="/" context={{}}>
      <Wallet {...defProps} />
    </StaticRouter>
  );
  await allOver();
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(3) usePrivate = false", () => {
  const props = {
    ...defProps,
    global: {
      ...globalInstance,
      usePrivate: false
    }
  };
  const renderer = TestRenderer.create(
    <StaticRouter location="/" context={{}}>
      <Wallet {...props} />
    </StaticRouter>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});
