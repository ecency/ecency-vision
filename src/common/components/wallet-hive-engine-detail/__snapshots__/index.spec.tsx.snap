// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default Render 1`] = `
<div
  className="wallet-hive"
>
  <div
    className="wallet-main"
  >
    <div
      className="wallet-info"
    >
      <div
        className="balance-row"
      >
        <div
          className="balance-info"
        >
          <div
            className="title"
          >
            Proof of Brain
          </div>
          <div
            className="description"
          >
            Proof of Brain are tradeable tokens that may be transferred at anytime.  POB can be staked in a process called powering up.
          </div>
        </div>
        <div
          className="balance-values"
        >
          <div
            className="amount"
          >
            <span>
              0 POB
            </span>
          </div>
        </div>
      </div>
      <div
        className="balance-row hive-power alternative"
      >
        <div
          className="balance-info"
        >
          <div
            className="title"
          >
            Brain Power
          </div>
          <div
            className="description"
          >
            Staked Proof of Brain earn more Proof of Brain for voting on posts. The more Staked Proof of Brain one holds, the more one can influence other’s rewards and earn rewards for accurate voting.
          </div>
        </div>
        <div
          className="balance-values"
        >
          <div
            className="amount"
          >
            0 POB
          </div>
        </div>
      </div>
      <div
        className="transaction-list"
      >
        <div
          className="transaction-list-header"
        >
          <h2>
             
            History
             
          </h2>
          <select
            className="form-control"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
               
              All History
            </option>
            <option
              value="transfers"
            >
              Transfers
            </option>
            <option
              value="market-orders"
            >
              Market Orders
            </option>
            <option
              value="interests"
            >
              Interests
            </option>
            <option
              value="stake-operations"
            >
              Stake operations
            </option>
            <option
              value="rewards"
            >
              Rewards
            </option>
          </select>
        </div>
        <div
          className="linear-progress"
        >
          <div
            className="bar bar1"
          />
          <div
            className="bar bar2"
          />
        </div>
      </div>
    </div>
    <div
      className="wallet-menu"
    >
      <a
        className="menu-item hive"
        href="/@user1/hive"
        onClick={[Function]}
      >
        <span
          className="title"
        >
          Hive
        </span>
        <span
          className="sub-title"
        >
          Wallet
        </span>
        <span
          className="platform-logo"
        >
          <svg
            height="100%"
            viewBox="0 0 512 512"
            width="100%"
          >
            <path
              d="M260.353,254.878,131.538,33.1a2.208,2.208,0,0,0-3.829.009L.3,254.887A2.234,2.234,0,0,0,.3,257.122L129.116,478.9a2.208,2.208,0,0,0,3.83-.009L260.358,257.113A2.239,2.239,0,0,0,260.353,254.878Zm39.078-25.713a2.19,2.19,0,0,0,1.9,1.111h66.509a2.226,2.226,0,0,0,1.9-3.341L259.115,33.111a2.187,2.187,0,0,0-1.9-1.111H190.707a2.226,2.226,0,0,0-1.9,3.341ZM511.7,254.886,384.9,33.112A2.2,2.2,0,0,0,382.99,32h-66.6a2.226,2.226,0,0,0-1.906,3.34L440.652,256,314.481,476.66a2.226,2.226,0,0,0,1.906,3.34h66.6a2.2,2.2,0,0,0,1.906-1.112L511.7,257.114A2.243,2.243,0,0,0,511.7,254.886ZM366.016,284.917H299.508a2.187,2.187,0,0,0-1.9,1.111l-108.8,190.631a2.226,2.226,0,0,0,1.9,3.341h66.509a2.187,2.187,0,0,0,1.9-1.111l108.8-190.631A2.226,2.226,0,0,0,366.016,284.917Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`(2) Render with converting HBD 1`] = `
<div
  className="wallet-hive"
>
  <div
    className="wallet-main"
  >
    <div
      className="wallet-info"
    >
      <div
        className="balance-row"
      >
        <div
          className="balance-info"
        >
          <div
            className="title"
          >
            Proof of Brain
          </div>
          <div
            className="description"
          >
            Proof of Brain are tradeable tokens that may be transferred at anytime.  POB can be staked in a process called powering up.
          </div>
        </div>
        <div
          className="balance-values"
        >
          <div
            className="amount"
          >
            <span>
              0 POB
            </span>
          </div>
        </div>
      </div>
      <div
        className="balance-row hive-power alternative"
      >
        <div
          className="balance-info"
        >
          <div
            className="title"
          >
            Brain Power
          </div>
          <div
            className="description"
          >
            Staked Proof of Brain earn more Proof of Brain for voting on posts. The more Staked Proof of Brain one holds, the more one can influence other’s rewards and earn rewards for accurate voting.
          </div>
        </div>
        <div
          className="balance-values"
        >
          <div
            className="amount"
          >
            0 POB
          </div>
        </div>
      </div>
      <div
        className="transaction-list"
      >
        <div
          className="transaction-list-header"
        >
          <h2>
             
            History
             
          </h2>
          <select
            className="form-control"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
               
              All History
            </option>
            <option
              value="transfers"
            >
              Transfers
            </option>
            <option
              value="market-orders"
            >
              Market Orders
            </option>
            <option
              value="interests"
            >
              Interests
            </option>
            <option
              value="stake-operations"
            >
              Stake operations
            </option>
            <option
              value="rewards"
            >
              Rewards
            </option>
          </select>
        </div>
        <div
          className="linear-progress"
        >
          <div
            className="bar bar1"
          />
          <div
            className="bar bar2"
          />
        </div>
      </div>
    </div>
    <div
      className="wallet-menu"
    >
      <a
        className="menu-item hive"
        href="/@user1/hive"
        onClick={[Function]}
      >
        <span
          className="title"
        >
          Hive
        </span>
        <span
          className="sub-title"
        >
          Wallet
        </span>
        <span
          className="platform-logo"
        >
          <svg
            height="100%"
            viewBox="0 0 512 512"
            width="100%"
          >
            <path
              d="M260.353,254.878,131.538,33.1a2.208,2.208,0,0,0-3.829.009L.3,254.887A2.234,2.234,0,0,0,.3,257.122L129.116,478.9a2.208,2.208,0,0,0,3.83-.009L260.358,257.113A2.239,2.239,0,0,0,260.353,254.878Zm39.078-25.713a2.19,2.19,0,0,0,1.9,1.111h66.509a2.226,2.226,0,0,0,1.9-3.341L259.115,33.111a2.187,2.187,0,0,0-1.9-1.111H190.707a2.226,2.226,0,0,0-1.9,3.341ZM511.7,254.886,384.9,33.112A2.2,2.2,0,0,0,382.99,32h-66.6a2.226,2.226,0,0,0-1.906,3.34L440.652,256,314.481,476.66a2.226,2.226,0,0,0,1.906,3.34h66.6a2.2,2.2,0,0,0,1.906-1.112L511.7,257.114A2.243,2.243,0,0,0,511.7,254.886ZM366.016,284.917H299.508a2.187,2.187,0,0,0-1.9,1.111l-108.8,190.631a2.226,2.226,0,0,0,1.9,3.341h66.509a2.187,2.187,0,0,0,1.9-1.111l108.8-190.631A2.226,2.226,0,0,0,366.016,284.917Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`(3) usePrivate = false 1`] = `
<div
  className="wallet-hive"
>
  <div
    className="wallet-main"
  >
    <div
      className="wallet-info"
    >
      <div
        className="balance-row"
      >
        <div
          className="balance-info"
        >
          <div
            className="title"
          >
            Proof of Brain
          </div>
          <div
            className="description"
          >
            Proof of Brain are tradeable tokens that may be transferred at anytime.  POB can be staked in a process called powering up.
          </div>
        </div>
        <div
          className="balance-values"
        >
          <div
            className="amount"
          >
            <span>
              0 POB
            </span>
          </div>
        </div>
      </div>
      <div
        className="balance-row hive-power alternative"
      >
        <div
          className="balance-info"
        >
          <div
            className="title"
          >
            Brain Power
          </div>
          <div
            className="description"
          >
            Staked Proof of Brain earn more Proof of Brain for voting on posts. The more Staked Proof of Brain one holds, the more one can influence other’s rewards and earn rewards for accurate voting.
          </div>
        </div>
        <div
          className="balance-values"
        >
          <div
            className="amount"
          >
            0 POB
          </div>
        </div>
      </div>
      <div
        className="transaction-list"
      >
        <div
          className="transaction-list-header"
        >
          <h2>
             
            History
             
          </h2>
          <select
            className="form-control"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
               
              All History
            </option>
            <option
              value="transfers"
            >
              Transfers
            </option>
            <option
              value="market-orders"
            >
              Market Orders
            </option>
            <option
              value="interests"
            >
              Interests
            </option>
            <option
              value="stake-operations"
            >
              Stake operations
            </option>
            <option
              value="rewards"
            >
              Rewards
            </option>
          </select>
        </div>
        <div
          className="linear-progress"
        >
          <div
            className="bar bar1"
          />
          <div
            className="bar bar2"
          />
        </div>
      </div>
    </div>
    <div
      className="wallet-menu"
    >
      <a
        className="menu-item hive"
        href="/@user1/hive"
        onClick={[Function]}
      >
        <span
          className="title"
        >
          Hive
        </span>
        <span
          className="sub-title"
        >
          Wallet
        </span>
        <span
          className="platform-logo"
        >
          <svg
            height="100%"
            viewBox="0 0 512 512"
            width="100%"
          >
            <path
              d="M260.353,254.878,131.538,33.1a2.208,2.208,0,0,0-3.829.009L.3,254.887A2.234,2.234,0,0,0,.3,257.122L129.116,478.9a2.208,2.208,0,0,0,3.83-.009L260.358,257.113A2.239,2.239,0,0,0,260.353,254.878Zm39.078-25.713a2.19,2.19,0,0,0,1.9,1.111h66.509a2.226,2.226,0,0,0,1.9-3.341L259.115,33.111a2.187,2.187,0,0,0-1.9-1.111H190.707a2.226,2.226,0,0,0-1.9,3.341ZM511.7,254.886,384.9,33.112A2.2,2.2,0,0,0,382.99,32h-66.6a2.226,2.226,0,0,0-1.906,3.34L440.652,256,314.481,476.66a2.226,2.226,0,0,0,1.906,3.34h66.6a2.2,2.2,0,0,0,1.906-1.112L511.7,257.114A2.243,2.243,0,0,0,511.7,254.886ZM366.016,284.917H299.508a2.187,2.187,0,0,0-1.9,1.111l-108.8,190.631a2.226,2.226,0,0,0,1.9,3.341h66.509a2.187,2.187,0,0,0,1.9-1.111l108.8-190.631A2.226,2.226,0,0,0,366.016,284.917Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>
</div>
`;
